farm/modeling/adaptive_model.py

```diff
@@ -631,139 +628,47 @@ def convert_from_transformers(cls, model_name_or_path, device, task_type, proces
 
         return adaptive_model
 
-    def convert_to_onnx(self, output_path, opset_version=11, optimize_for=None):
-        """
-        Convert a PyTorch AdaptiveModel to ONNX.
-
-        The conversion is trace-based by performing a forward pass on the model with a input batch.
-
-        :param output_path: model dir to write the model and config files
-        :type output_path: Path
-        :param opset_version: ONNX opset version
-        :type opset_version: int
-        :param optimize_for: optimize the exported model for a target device. Available options
-                             are "gpu_tensor_core" (GPUs with tensor core like V100 or T4),
-                             "gpu_without_tensor_core" (most other GPUs), and "cpu".
-        :type optimize_for: str
-        :return:
-        """
-        if type(self.prediction_heads[0]) is not QuestionAnsweringHead:
-            raise NotImplementedError
+    @classmethod
+    def convert_to_onnx(cls, model_name, output_path, task_type, opset_version=11):
+        language_model_class = LanguageModel.get_language_model_class(model_name)
+        if language_model_class not in ["Bert", "Roberta"]:
+            raise Exception("The current ONNX conversion only support 'BERT' & 'RoBERTa' models")
+
+        task_type_to_pipeline_map = {
+            "question_answering": "question-answering",
+            "embeddings": "feature-extraction",
+            "ner": "ner"
+        }
 
-        tokenizer = Tokenizer.load(
-            pretrained_model_name_or_path="deepset/bert-base-cased-squad2"
+        convert(
```

Comment:
Does this convert function include the same onnx optimizations as we had before? We saw quite big improvements due to them in our benchmarks ..

Reply:
Many optimizations are now part of ONNXRuntime and are applied automatically when an inference session is created. 

I added two additional optimizations:
1. Conversion to float16: Earlier we were using optimization scripts from `farm.conversion` but now they're available in the `onnxruntime_tools` Python package. The conversion now uses this package and a new parameter `convert_to_float16` is added in `convert_to_onnx()` to apply this optimization.
2. Quantization: An option is added for the quantization of floating-point numbers to integers provided by the transformers that can be used by setting the `quantize` parameter in `convert_to_onnx()`.